core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -554,10 +559,15 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   /**
    * Replays the events in the specified log file and returns information about the associated
    * application. Return `None` if the application ID cannot be located.
+   *
+   * Depending on the eventsFilter, return values may not be populated or get returned as `None`
+   * because the events carrying the needed data are filtered out and therefore not replayed.
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: ReplayEventsFilter = SELECT_ALL_FILTER)
+  : (Option[(FsApplicationAttemptInfo, FsApplicationEnvACLInfo)]) = {
```

Comment:
What about returning the `ApplicationEventListener` instead of creating this new type (`FsApplicationEnvACLInfo`) for the info that exists there?